File: apps/mock-runner/pkg/models/enums/sandbox_state.go
// Copyright 2025 Daytona Platforms Inc.
// SPDX-License-Identifier: AGPL-3.0

package enums

type SandboxState string

const (
	SandboxStateCreating        SandboxState = "creating"
	SandboxStateRestoring       SandboxState = "restoring"
	SandboxStateDestroyed       SandboxState = "destroyed"
	SandboxStateDestroying      SandboxState = "destroying"
	SandboxStateStarted         SandboxState = "started"
	SandboxStateStopped         SandboxState = "stopped"
	SandboxStateStarting        SandboxState = "starting"
	SandboxStateStopping        SandboxState = "stopping"
	SandboxStateResizing        SandboxState = "resizing"
	SandboxStateError           SandboxState = "error"
	SandboxStateUnknown         SandboxState = "unknown"
	SandboxStatePullingSnapshot SandboxState = "pulling_snapshot"
)

func (s SandboxState) String() string {
	return string(s)
}

File: libs/sdk-java/src/main/java/io/daytona/sdk/model/SandboxState.java
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

package io.daytona.sdk.model;

/**
 * Lifecycle state of a Sandbox.
 *
 * <p>This is the SDK-facing mirror of the api-client enum. It exists so users
 * never need to import from the api-client package.
 */
public enum SandboxState {
    CREATING("creating"),
    RESTORING("restoring"),
    DESTROYED("destroyed"),
    DESTROYING("destroying"),
    STARTED("started"),
    STOPPED("stopped"),
    STARTING("starting"),
    STOPPING("stopping"),
    ERROR("error"),
    BUILD_FAILED("build_failed"),
    PENDING_BUILD("pending_build"),
    BUILDING_SNAPSHOT("building_snapshot"),
    UNKNOWN("unknown"),
    PULLING_SNAPSHOT("pulling_snapshot"),
    ARCHIVED("archived"),
    ARCHIVING("archiving"),
    RESIZING("resizing"),
    SNAPSHOTTING("snapshotting"),
    FORKING("forking");

    private final String value;

    SandboxState(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }

    /** Convert to the api-client enum (used internally by the SDK). */
    public io.daytona.api.client.model.SandboxState toApiClient() {
        return io.daytona.api.client.model.SandboxState.fromValue(value);
    }
}
